modules/auxiliary/gather/shodan_search.rb

```diff
@@ -157,11 +161,10 @@ def run
     )
 
     # Organize results and put them into the table and database
-    p = 1
+    page = 0
     regex = datastore['REGEX'] if datastore['REGEX']
-    while p <= maxpage
-      break if p > maxpage
-      results[p]['matches'].each do |host|
+    while page < maxpage
+      results[page]['matches'].each do |host|
```

Comment:
Instead of defining and then manually incrementing `page` here, this could just operate on `results.each`. With the other changes I suggested, there won't be any page results where `page_results['matches']` is nil.

That would make this something like:

```ruby
    results.each do |page_results|
```

At that point you don't even need `page`, don't have to define it, don't have to increment it.